carma-messenger-core/cpp_message/src/Map_Message.cpp

```diff
@@ -0,0 +1,908 @@
+                                {
+                                    entry.connecting_lane.maneuver_exists = true;
+                                    entry.connecting_lane.maneuver.allowed_maneuvers = *ct->connectingLane.maneuver->buf;
+                                }
+                                else
+                                {
+                                    entry.connecting_lane.maneuver_exists=false;
+                                }
+                                
+                                //Remote Intersection
+                                if(ct->remoteIntersection)
+                                {
+                                    entry.remote_intersection_exists = true;
+                                    
+                                    //Remote Intersection Region
+                                    if(ct->remoteIntersection->region)
+                                    {
+                                        entry.remote_intersection.region_exists = true;
+                                        entry.remote_intersection.region = *ct->remoteIntersection->region;
+                                    }
+                                    else
+                                    {
+                                        entry.remote_intersection.region_exists = false;
+                                    }
+
+                                    entry.remote_intersection.id = ct->remoteIntersection->id;
+
+                                }
+                                else
+                                {
+                                    entry.remote_intersection_exists = false;
+                                }
+
+                                //Signal Group
+                                if(ct->signalGroup)
+                                {
+                                    entry.signal_group_exists = true;
+                                    entry.signal_group = *ct->signalGroup;
+                                }
+                                else
+                                {
+                                    entry.signal_group_exists = false;
+                                }
+
+                                //User Class
+                                if (ct->userClass)
+                                {
+                                    entry.user_class_exists = true;
+                                    entry.user_class = *ct->userClass;
+                                }
+
+                                ln.connects_to.connect_to_list.push_back(entry);
+
+                            }
+
+                        }
+                        else
+                        {
+                            ln.egress_approach_exists = false;
+                        }
+
+                        new_intersection.lane_set.lane_list.push_back(ln);
+                    }
+
+                    if(map_msg_intersections->name)
```

Comment:
Can we move the order in which the fields are parsed so that it resembles the message definition? Just to make it easier to read. For example intersections.name is the first field that appears in the intersection definition.
Not completely necessary, but it would help ensure we didn't miss a field